### PR TITLE
Initial support for generating go protos

### PR DESCRIPTION
**Pull request by shan4993**

### CODE REVIEW

File: t/proto_template/scripts/generate.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
```
Comment (sethvargo) [inline t/proto_template/scripts/generate.sh:1]:
```suggestion
#!/usr/bin/env bash
set -eEuo pipefail
```

File: t/proto_template/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  brew install protobuf # for MacOS users
+  apt install -y protobuf-compiler # for Linux users
+```
+
+### Go
+
+Ensure you have go installed, if not run the following
+
+```
+  brew install go
```
Comment (pdewilde) [inline t/proto_template/README.md:19]: Should also have linux instructions

File: t/proto_template/scripts/generate.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  # Create the new directory if it doesn't exist
+  if [ ! -d $1 ]; then
+    mkdir -p $1
+  fi
+
+  # Check if the go.mod exists
+  if [ ! -f "$1/go.mod" ]; then
+    # Create the file
+    cd $1
+    if ! go mod init github.com/kiyani-org/proto-solution
```
Comment (drevell) [inline t/proto_template/scripts/generate.sh:48]: Is "kiyani-org" supposed to be some placeholder value instead?

File: t/proto_template/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# limitations under the License.
+
+apiVersion: 'cli.abcxyz.dev/v1alpha1'
+kind: 'Template'
+
+desc:
+  'A template that sets up a protos directory along with a script to generate language specific protos.'
+inputs:
+  - name: 'github_org_name'
+    desc: 'The name of te github org the repo will belong to'
```
Comment (drevell) [inline t/proto_template/spec.yaml:22]: "te"

File: t/proto_template/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  - name: 'github_repo_name'
+    desc: 'The name of the github repo'
+steps:
+  - desc: 'Include some files and directories'
+    action: 'include'
+    params:
+      paths: ['protos', 'scripts']
+  - desc: 'Replace "github_org_name" with '
```
Comment (drevell) [inline t/proto_template/spec.yaml:30]: I think maybe you intended to add more words in the desc, it kinda ends unexpectedly

**Review by drevell — CHANGES_REQUESTED** [review]

File: t/proto_template/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# Context
+
+You must have various tools installed in order to generate protos. The tool does not distinguish between language types and instead seeks to generate protos for a variety of supported languages. Currently only go is supported. Node will be supported in the near future.
```
Comment (pdewilde) [inline t/proto_template/README.md:3]:
```suggestion
Every language has its own tools for generating protos. The tool will generate protos for a variety of supported languages. Currently only go is supported. Node will be supported in the near future.
```

File: t/proto_template/README.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+## Tools
+
+Install `protoc`, a compiler for generating language specific protos based on .proto files. 
+
+```
+  brew install protobuf # for MacOS users
+  apt install -y protobuf-compiler # for Linux users
```
Comment (pdewilde) [inline t/proto_template/README.md:11]: Nit: apt is only for debian-based distros, other linux distros have other package managers (dnf/yum/pacman)

File: t/proto_template/README.md (hunk 2 above)
Comment (sethvargo) [reply t/proto_template/README.md:19]: https://docs.brew.sh/Homebrew-on-Linux 🙃

File: t/proto_template/scripts/generate.sh (hunk 3 above)
Comment (shan4993) [reply t/proto_template/scripts/generate.sh:49]: ack sorry about that

**Review by sethvargo — APPROVED** [review]

**Review by drevell — APPROVED** [review]

**Pull request merged**